InvoiceMaker/InvoiceMaker/Models/Invoice.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Web;
+
+namespace InvoiceMaker.Models
+{
+    public class Invoice
+    {
+        public Invoice()
+        {
+
+        }
+        public Invoice(string invoiceNumber)
+        {
+            InvoiceNumber = invoiceNumber;
+            LineItems = new List<ILineItem>();
+            Status = InvoiceStatus.Open;
+        }
+
+        public Invoice(string invoiceNumber, InvoiceStatus status)
+            : this(invoiceNumber)
+        {
+            Status = status;
+        }
+
+        public Invoice(string invoiceNumber, InvoiceStatus status, int clientId)
+            : this(invoiceNumber)
+        {
+            Status = status;
+            ClientId = clientId;
+        }
+
+        public void FinalizeInvoice()
+        {
+            if (Status == InvoiceStatus.Open)
+            {
+                Status = InvoiceStatus.Finalized;
+            }
+        }
+
+        public void CloseInvoice()
+        {
+            if (Status == InvoiceStatus.Finalized)
+            {
+                Status = InvoiceStatus.Closed;
+            }
+        }
+
+        public void AddWorkLineItem(WorkDone workDone)
+        {
+            LineItems.Add(new WorkLineItem(workDone));
+        }
+
+        public void AddFeeLineItem(string description, decimal amount, DateTimeOffset when)
+        {
+            LineItems.Add(new FeeLineItem(description, amount, when));
+        }
+
+        public int Id { get; set; }
+        [Column("Status")]
+        public InvoiceStatus Status { get; set; }
+        [Column("InvoiceNumber")]
+        public string InvoiceNumber { get; set; }
```

Comment:
`Column` attributes are only necessary if the column name needs to be different from the property name.